File: Question 1-10/Q7/Q7.java
import java.util.*;
public class Q7
{
    int findSingle(int arr[])
    {
        int res = 0;
        for(int i=0;i<arr.length;i++)
            res = res ^ arr[i];
        return res;
    }
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        Q7 obj = new Q7();
        int n = sc.nextInt();
        int arr[] = new int[n];
        for(int i=0;i<n;i++)
            arr[i] = sc.nextInt();
            
        System.out.println(obj.findSingle(arr));
    }
}

File: Question 1-10/Q7/Q7.c
#include <stdio.h>
#include <stdlib.h>
int findSingle(int * arr, int n){
    int i, res = 0;
    for(i=0;i<n;i++)
        res = res ^ arr[i];
    return res;
}

int main(void){
    int n, *arr, i;
    scanf("%d", &n);
    arr = (int *)malloc(n*sizeof(int));

    for(i=0;i<n;i++)
        scanf("%d", &arr[i]);

    printf("%d\n", findSingle(arr, n));
    return 0;
}
